pkg/controller/gitopsconfig/jobmonitor.go

```diff
@@ -157,6 +158,10 @@ func (job *jobmonitor) watchjob(instanceName, instanceNamespace string) {
 		time.Sleep(10 * time.Second)
 		jobState, err := job.checkResult()
 		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				log.Info("The job has been deleted")
+				break
```

Comment:
Before, when there's been a deployment, the job was being watched for status and in the meantime one decided to remove the namespace, the nonexistent job was being watched for status without an end.

Before:
<details><summary><code>Logs from eunomia-operator</code></summary><br><pre>
{"level":"error","ts":1576746595.449615,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746615.454666,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746615.454728,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746635.4603698,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746635.460654,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746655.463279,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746655.463386,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746675.481663,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746675.481712,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746695.5009239,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746695.50108,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746715.511931,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746715.512026,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
{"level":"info","ts":1576746735.515382,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"error","ts":1576746735.515475,"logger":"controller_gitopsconfig","msg":"Error in GetJob for CheckResult","error":"Job.batch \"gitopsconfig-hello-world-yaml-nd2iir\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).checkResult\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:191\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.(*jobmonitor).watchjob\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:158\ngithub.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig.watchJobForStatus\n\t/Users/sbar/go/src/github.com/KohlsTechnology/eunomia/pkg/controller/gitopsconfig/jobmonitor.go:143"}
...
</pre></details>

Now:
<details><summary><code>Logs from eunomia-operator</code></summary><br><pre>
{"level":"info","ts":1576747151.492175,"logger":"controller_gitopsconfig","msg":"retrying the GetJob for CheckResult"}
{"level":"info","ts":1576747151.492248,"logger":"controller_gitopsconfig","msg":"Retry of GetJob for CheckResult unsuccessful: Job.batch \"gitopsconfig-hello-world-yaml-fjxaon\" not found"}
{"level":"info","ts":1576747151.492257,"logger":"controller_gitopsconfig","msg":"The job has been deleted"}
</pre></details>